sentry-android-core/src/main/java/io/sentry/android/core/ActivityFramesState.java

```diff
@@ -0,0 +1,82 @@
+package io.sentry.android.core;
+
+import android.app.Activity;
+import android.util.SparseIntArray;
+import androidx.core.app.FrameMetricsAggregator;
+import io.sentry.protocol.MeasurementValue;
+import java.util.HashMap;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+final class ActivityFramesState {
+
+  private static final @NotNull ActivityFramesState instance = new ActivityFramesState();
+
+  private final @NotNull FrameMetricsAggregator frameMetricsAggregator =
+      new FrameMetricsAggregator();
+
+  private ActivityFramesState() {}
+
+  static @NotNull ActivityFramesState getInstance() {
+    return instance;
+  }
+
+  void addActivity(final @NotNull Activity activity) {
+    frameMetricsAggregator.add(activity);
+  }
+
+  void removeActivity(final @NotNull Activity activity) {
+    frameMetricsAggregator.remove(activity);
+  }
+
+  @Nullable
+  Map<String, @NotNull MeasurementValue> getMetrics() {
```

Comment:
`l`: What about returning a data class with total, slow and frozen frames here instead? The `PerformanceAndroidEventProcessor` could then convert it to the correct format.

Reply:
I thought about it, but we just need to add these measurements to the current transactions.measurements, nothing more, its not used outside of this scope, it just sounded over-engineering, like, creating a new object and moving back to a Map anyway